code/modules/religion/sects/shadow_sect.dm

```diff
@@ -345,10 +381,466 @@
+		return FALSE
+	if(!((sect.light_power <= -6 - 5 * sect.grand_ritual_level) || (sect.light_reach >= 8 + 7.5 * sect.grand_ritual_level)))
+		to_chat(user, span_warning("You need to strengthen the shadows before you can begin the ritual. Expand shadows to their limits."))
+		return FALSE
+	if(sect.active_obelisk_number < 5 * (sect.grand_ritual_level + 1))
+		to_chat(user, span_warning("You need to anchor the shadows to this reality. You need [5 * (sect.grand_ritual_level + 1)] active obelisks."))
+		return FALSE
+	if(!can_afford(user))
+		return FALSE
+	var/turf/T = get_turf(religious_tool)
+	if(!T.is_holy())
+		to_chat(user, span_warning("The altar can only function in a holy area!"))
+		return FALSE
+	if(!GLOB.religious_sect.altar_anchored)
+		to_chat(user, span_warning("The altar must be secured to the floor if you wish to perform the rite!"))
+		return FALSE
+	spawn()
+		handle_obelisks()
+	return ..()
+
+/datum/religion_rites/grand_ritual_one/invoke_effect(mob/living/user, atom/religious_tool)
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	if(sect.active_obelisk_number < 5 * (sect.grand_ritual_level + 1))
+		to_chat(user, span_warning("Your obelisk have been destroed, destabilising the ritual!. You need to gather your strength and try again."))
+		sect.adjust_favor(-1 * favor_cost)
+		return FALSE
+	sect.grand_ritual_level = 1
+	for(var/obj/structure/destructible/religion/shadow_obelisk/obelisk in sect.obelisks)
+		obelisk.transform_obelisk()
+	for(var/mob/living/carbon/human/M in GLOB.mob_list)
+		if(isshadow(M))
+			M.heal_overall_damage(100, 100, 200)
+			var/datum/species/shadow/spiec = M.dna.species
+			spiec.change_hearts_ritual(M)
+	sect.rites_list -= /datum/religion_rites/grand_ritual_one
+	sect.rites_list += /datum/religion_rites/grand_ritual_two
+	return ..()
+
+/datum/religion_rites/grand_ritual_one/proc/handle_obelisks()
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	for(var/mob/living/M in GLOB.mob_list)
+		if(isshadow(M))
+			to_chat(M, span_userdanger("You feel pull towards the obelisks, you feel like it woudl be safer near them."))
+		to_chat(M, span_notice("You see shadows flicker in corner of your eye."))
+	sleep(50)
+	sect.grand_ritual_in_progress = TRUE
+	for(var/obj/structure/destructible/religion/shadow_obelisk/obelisk in sect.obelisks)
+		if(obelisk.anchored)
+			obelisk.set_light(4, -10, DARKNESS_INVERSE_COLOR)
+	sleep(300)
+	for(var/obj/structure/destructible/religion/shadow_obelisk/obelisk in sect.obelisks)
+		if(obelisk.anchored)
+			obelisk.set_light(sect.light_reach, sect.light_power, DARKNESS_INVERSE_COLOR)
+	sect.grand_ritual_in_progress = FALSE
+
+
+/datum/religion_rites/grand_ritual_two
+	name = "Grand ritual: Infusing shadows"
+	desc = "Start giving shadows a form in physical world. This will let them heal the vounds of their kin and protect them from sight or harm."
```

Comment:
```suggestion
	desc = "Start giving shadows a form in physical world. This will let them heal the wounds of their kin and protect them from sight or harm."
```